[pt-br] Add docs/tasks/configure-pod-container/security-context.md (#35118)
kubernetes/website: content/pt-br/docs/tasks/configure-pod-container/security-context.md

e a descrição do campo `VolumeCapability.MountVolume.volume_mount_group`
na [especificação CSI](https://github.com/container-storage-interface/spec/blob/master/spec.md#createvolume)
para maiores informações.

## Defina o contexto de segurança para um Contêiner

Para especificar configurações de segurança a um Contêiner, inclua o campo `securityContext` no manifesto do Contêiner. O campo `securityContext` é um objeto
[SecurityContext](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#securitycontext-v1-core).
Configurações de segurança que você especifica para um contêiner se aplicam apenas ao contêiner individual, e substituem as configurações feitas no nível do Pod quando
há sobreposição. As configurações de Contêiner não afetam os Volumes do Pod.

Aqui está o arquivo de configuração para um pod que possui um contêiner. Ambos, o Pod
e o Contêiner, tem um campo `securityContext`:

{{< codenew file="pods/security/security-context-2.yaml" >}}

Crie o Pod:

```shell
kubectl apply -f https://k8s.io/examples/pods/security/security-context-2.yaml
```

Verifique se o Contêiner do Pod está executado:

```shell
kubectl get pod security-context-demo-2
```

> Abra um shell no Contêiner em execução:

Reviewer: ```suggestion
Abra um shell no contêiner em execução:
```

```shell
kubectl exec -it security-context-demo-2 -- sh
```

No seu shell, liste os processos em execução:

```shell
ps aux
```

A saída mostra que os processos estão em execução como usuário 2000. Este é o valor

> do `runAsUser` especificado para o Contêiner. Ele sobrepõe o valor 1000 que é especificado para o Pod.

Reviewer: ```suggestion
do campo `runAsUser` especificado para o contêiner. Ele toma precedência sobre o valor 1000 que é especificado para o Pod.
```

```
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
2000         1  0.0  0.0   4336   764 ?        Ss   20:36   0:00 /bin/sh -c node server.js
2000         8  0.1  0.5 772124 22604 ?        Sl   20:36   0:00 node server.js
...
```

Saia do seu shell:

```shell
exit
```

## Defina recursos para um Contêiner

> Com [recursos Linux](https://man7.org/linux/man-pages/man7/capabilities.7.html),

Reviewer: I'd keep capabilities here. You could stylize it as italic (i.e. _capabilities_) to give the reader the sense that this comes from a foreign language.

> Você pode conceder certos privilégios a um processo, sem conceder todos os privilégios
> do usuário root. Para adicionar ou remover recursos Linux para um Contêiner, inclua o campo `capabilities` na seção `securityContext` do manifesto do Contêiner.
> 
> Primeiro, veja o que acontece quando você não inclui um campo `capabilities`.
> Aqui está o arquivo de configuração que não adiciona ou remove nenhum recurso do Contêiner:

Reviewer: ```suggestion
Você pode conceder certos privilégios a um processo, sem conceder todos os privilégios
do usuário root. Para adicionar ou remover recursos Linux para um contêiner, inclua o campo `capabilities` na seção `securityContext` do manifesto do contêiner.

Primeiro, veja o que acontece quando você não inclui um campo `capabilities`.
Aqui está o arquivo de configuração que não adiciona ou remove nenhum recurso do contêiner:
```

{{< codenew file="pods/security/security-context-3.yaml" >}}

Crie o Pod:

```shell
kubectl apply -f https://k8s.io/examples/pods/security/security-context-3.yaml
```

> Verifique se o Contêiner do Pod está executando:

Reviewer: ```suggestion
Verifique que o contêiner do Pod está executando:
```

```shell
kubectl get pod security-context-demo-3
```

> Abra um shell dentro do Contêiner em execução:

Reviewer: ```suggestion
Abra um shell dentro do contêiner em execução:
```

```shell
kubectl exec -it security-context-demo-3 -- sh
```

No seu shell, liste os processos em execução:

```shell
ps aux
```

> A saída mostra os IDs dos processos (PIDs) para o Contêiner:

Reviewer: ```suggestion
A saída mostra os IDs dos processos (PIDs) para o contêiner:
```

```
USER  PID %CPU %MEM    VSZ   RSS TTY   STAT START   TIME COMMAND
root    1  0.0  0.0   4336   796 ?     Ss   18:17   0:00 /bin/sh -c node server.js
root    5  0.1  0.5 772124 22700 ?     Sl   18:17   0:00 node server.js
```

No seu shell, veja o status para o processo 1:

```shell
cd /proc/1
cat status
```

A saída mostra o bitmap de recursos para o processo:

```
...
CapPrm:	00000000a80425fb
CapEff:	00000000a80425fb
...
```

Anote o bitmap dos recursos e depois saia do seu shell:

```shell
exit
```

> Em seguida, execute um Contêiner que seja o mesmo que o Contêiner anterior, exceto
> que possui recursos adicionais definidos.
> 
> Aqui está o arquivo de configuração para um Pod que executa um Contêiner. A configuração
> adiciona os recursos `CAP_NET_ADMIN` e `CAP_SYS_TIME`:

Reviewer: ```suggestion
Em seguida, execute um contêiner que seja o mesmo que o contêiner anterior, porém com recursos adicionais definidos.

Aqui está o arquivo de configuração para um Pod que executa um contêiner. A configuração
adiciona os recursos `CAP_NET_ADMIN` e `CAP_SYS_TIME`:
```

{{< codenew file="pods/security/security-context-4.yaml" >}}

Crie o Pod:

```shell
kubectl apply -f https://k8s.io/examples/pods/security/security-context-4.yaml
```

> Abra um shell no Contêiner em execução:

Reviewer: ```suggestion
Abra um shell no contêiner em execução:
```

```shell
kubectl exec -it security-context-demo-4 -- sh
```

No seu shell, veja os recursos para o processo 1:

```shell
cd /proc/1
cat status
```

A saída mostra os recursos de bitmap para o processo:

```
...
CapPrm:	00000000aa0435fb
CapEff:	00000000aa0435fb
...
```

> Compare os recursos dos dois Contêineres:

Reviewer: ```suggestion
Compare os recursos dos dois contêineres:
```

```
00000000a80425fb
00000000aa0435fb
```

> No bitmap de capacidade do primeiro contêiner, os bits 12 e 25 estão vazios. No segundo Contêiner,

Reviewer: ```suggestion
No bitmap de _capabilities_ do primeiro contêiner, os bits 12 e 25 estão vazios. No segundo contêiner,
```

os bits 12 e 25 estão definidos. O bit 12 é `CAP_NET_ADMIN`, e o bit 25 é `CAP_SYS_TIME`.
Veja [capability.h](https://github.com/torvalds/linux/blob/master/include/uapi/linux/capability.h)

> para definições das constantes de capacidade.

Reviewer: ```suggestion
para definições das constantes de _capabilities_.
```

{{< note >}}

> As constantes de capacidade do Linux têm o formato `CAP_XXX`.
> Mas quando você listar os recursos no manifesto do seu contêiner, você deve
> omitir a parte `CAP_` da constante.
> Por exemplo, para adicionar `CAP_SYS_TIME`, inclua `SYS_TIME` na sua lista de recursos.

Reviewer: ```suggestion
As constantes de _capabilities_ do Linux têm o formato `CAP_XXX`.
Mas quando você listar os recursos no manifesto do seu contêiner, você deve
omitir a parte `CAP_` da constante.
Por exemplo, para adicionar `CAP_SYS_TIME`, inclua `SYS_TIME` na sua lista de recursos.
```

{{< /note >}}

> ## Defina o perfil do `Seccomp` para um Contêiner
> 
> Para definir o perfil do `Seccomp` para um Contêiner, inclua o campo `seccompProfile` na seção `securityContext` do seu Pod ou manifesto do Contêiner. 
> O campo `seccompProfile` é um objeto
> [SeccompProfile](/docs/reference/generated/kubernetes-api/{{< param "version">}}/#seccompprofile-v1-core) consistindo de `type` e `localhostProfile`.
> As opções válidas para `type` incluem `RuntimeDefault`, `Unconfined`, e
> `Localhost`. O `localhostProfile` só deve ser definido se `type: Localhost`. 
> Que indica o caminho do perfil pré-configurado no nó, relativo a localização do perfil de `Seccomp` do Kubelet 
> (configurado com o flag `--root-dir`).
> 
> Aqui está um exemplo que define o perfil `Seccomp` ao perfil padrão do nó do contêiner runtime:

Reviewer: ```suggestion
## Defina o perfil do Seccomp para um Contêiner

Para definir o perfil do Seccomp para um contêiner, inclua o campo `seccompProfile` na seção `securityContext` do seu Pod ou manifesto do contêiner. 
O campo `seccompProfile` é um objeto
[SeccompProfile](/docs/reference/generated/kubernetes-api/{{< param "version">}}/#seccompprofile-v1-core) consistindo de `type` e `localhostProfile`.
As opções válidas para `type` incluem `RuntimeDefault`, `Unconfined`, e
`Localhost`. O campo `localhostProfile` só deve ser definido quando `type: Localhost`. 
Este campo indica o caminho do perfil pré-configurado no nó, relativo à localização do perfil de `Seccomp` do kubelet 
(configurado com a flag `--root-dir`).

Aqui está um exemplo que define o perfil Seccomp ao perfil padrão do nó do agente de execução do contêiner:
```

```yaml
...
securityContext:
  seccompProfile:
    type: RuntimeDefault
```

> Aqui está um exemplo que define o perfil `Seccomp` para um arquivo pré-configurado em

Reviewer: ```suggestion
Aqui está um exemplo que define o perfil Seccomp para um arquivo pré-configurado em
```

`<kubelet-root-dir>/seccomp/my-profiles/profile-allow.json`:

```yaml
...
securityContext:
  seccompProfile:
    type: Localhost
    localhostProfile: my-profiles/profile-allow.json
```

> ## Atribua rótulos `SELinux` a um Contêiner
> 
> Para Atribuir rótulos `SELinux` a um Contêiner, inclua o campo `seLinuxOptions` na seção `securityContext` do seu Pod ou manifesto do Contêiner. O campo
> `seLinuxOptions` é um objeto
> [SELinuxOptions](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#selinuxoptions-v1-core). 
> Aqui está um exemplo que aplica um nível `SELinux`:

Reviewer: ```suggestion
## Atribua rótulos SELinux a um contêiner

Para atribuir rótulos SELinux a um contêiner, inclua o campo `seLinuxOptions` na seção `securityContext` do seu Pod ou manifesto do contêiner. O campo
`seLinuxOptions` é um objeto
[SELinuxOptions](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#selinuxoptions-v1-core). 
Aqui está um exemplo que aplica um nível SELinux:
```

```yaml
...
securityContext:
  seLinuxOptions:
    level: "s0:c123,c456"
```

{{< note >}}
Para atribuir rótulos SELinux, o módulo de segurança SELinux deve ser carregado no sistema operacional host.
{{< /note >}}

> 

Reviewer: This section is missing the "Efficient SELinux volume relabeling" subsection.

## Discussão

> O contexto de segurança de um Pod se aplica aos Contêineres do Pod e também aos Volumes do Pod quando aplicável. Especificamente `fsGroup` e `seLinuxOptions` são
> aplicados a Volumes da seguinte maneira:
> 
> * `fsGroup`: Volumes que aceitam o gerenciamento de propriedade, são
>   modificados para o proprietário e permissão de escrita ao GID especificado no `fsGroup`. Veja o
>   [Documento de desenho de gerenciamento de propriedade](https://git.k8s.io/design-proposals-archive/storage/volume-ownership-management.md)
>   para mais detalhes.
> 
> * `seLinuxOptions`: Volumes que suportam rotulagem SELinux são re-rotulados para serem acessíveis pelo rótulo
>   especificado em `seLinuxOptions`. Geralmente você apenas
>   precisa definir a seção `level`. Isso define a
>   [Segurança de múltiplas categorias (MCS)](https://selinuxproject.org/page/NB_MLS),
>   um rótulo fornecido a todos os Contêineres no pod e aos volumes.
> 
> {{< warning >}}
> Depois de especificar um rótulo MCS para um Pod, todos os Pods com o mesmo rótulo podem acessar o Volume.
> Se você precisar de proteção Inter-Pod, você deve atribuir um rótulo único MCS a cada Pod.
> {{< /warning >}}

Reviewer: ```suggestion
O contexto de segurança de um Pod se aplica aos contêineres do Pod e também aos Volumes do Pod quando aplicável. Especificamente `fsGroup` e `seLinuxOptions` são
aplicados a volumes da seguinte maneira:

* `fsGroup`: Volumes que aceitam o gerenciamento de propriedade, são
  modificados para o proprietário e permissão de escrita ao GID especificado no `fsGroup`. Veja o
  [documento de design de gerenciamento de propriedade](https://git.k8s.io/design-proposals-archive/storage/volume-ownership-management.md)
  para mais detalhes.

* `seLinuxOptions`: Volumes que suportam rotulagem SELinux são re-rotulados para serem acessíveis pelo rótulo
  especificado em `seLinuxOptions`. Geralmente você
  precisa somente definir a seção `level`. Isso define a
  [Segurança de múltiplas categorias (_Multi-Category Security_, ou MCS)](https://selinuxproject.org/page/NB_MLS),
  um rótulo fornecido a todos os Contêineres no pod e aos volumes.

{{< warning >}}
Depois de especificar um rótulo MCS para um Pod, todos os Pods com o mesmo rótulo podem acessar o volume.
Se você precisar de proteção entre Pods, você deve atribuir um rótulo único MCS a cada Pod.
{{< /warning >}}
```

---
update_date: 2022-07-18
origin_version: 1.24
contributors: DonatoHorn

> reviewers: 
> - erictune
> - mikedanese

Reviewer 2: ```suggestion
```

Reviewer: @DonatoHorn this section still needs to be removed, since the listed reviewers do not review content in Brazilian Portuguese.

- thockin

> title: Configurando um Contexto de Segurança Para um Pod ou Contêiner

Reviewer 2: If writing the translated term “Contêiner”, don't capitalize the first letter.

---

> reviewers: 
> - erictune
> - mikedanese

Reviewer: ```suggestion
```

This section must still be removed.

> title: Configurando um Contexto de Segurança Para um Pod ou Contêiner

Reviewer 3: "para" instead of "Para"

content_type: task
weight: 80
---

<!-- overview -->

Um contexto de segurança define configurações de controle de privilégio e acesso para
um Pod ou Contêiner. As configurações de contexto de segurança incluem, mas não estão limitadas, a:

* Controle de acesso discricionário: Permissão para acessar um objeto, como um arquivo, é baseada em
  [ID de usuário (UID) e ID de grupo (GID)](https://wiki.archlinux.org/index.php/users_and_groups).

* [Segurança aprimorada Linux (SELinux)](https://pt.wikipedia.org/wiki/SELinux):

>   os objetos recebem rótulos de segurança.

Reviewer 3: "Os objetos" instead of "os objetos" The capital is to follow the other items

`feature gate` esteja habilitada. Nesse caso, contanto que o Kubernetes não realize nenhuma
mudança de propriedade e permissão, `fsGroupChangePolicy` não entra em vigor e
conforme especificado por CSI, é esperado que o driver monte o volume com `fsGroup` fornecido, resultando em um volume que pode ser lido/gravado pelo
`fsGroup`.

## Defina o contexto de segurança para um Contêiner

Para especificar configurações de segurança a um Contêiner, inclua o campo `securityContext` no manifesto do Contêiner. O campo `securityContext` é um objeto
[SecurityContext](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#securitycontext-v1-core).
Configurações de segurança que você especifica para um contêiner se aplicam apenas ao contêiner individual, e substituem as configurações feitas no nível do Pod quando
há sobreposição. As configurações de contêiner não afetam os volumes do Pod.

Aqui está o arquivo de configuração para um Pod que possui um contêiner. Ambos o Pod
e o contêiner têm um campo `securityContext`:

{{< codenew file="pods/security/security-context-2.yaml" >}}

Crie o Pod:

```shell
kubectl apply -f https://k8s.io/examples/pods/security/security-context-2.yaml
```

> Verifique que o contêiner do Pod está rodando:

Reviewer 3: What do you think of "em execução" like line 232 and 238

```shell
kubectl get pod security-context-demo-2
```

Abra um shell no contêiner em execução:

```shell
kubectl exec -it security-context-demo-2 -- sh
```

No seu shell, liste os processos em execução:

```shell
ps aux
```

A saída mostra que os processos estão em execução como usuário 2000. Este é o valor
do campo `runAsUser` especificado para o contêiner. Ele toma precedência sobre o valor 1000 que é especificado para o Pod.

```
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
2000         1  0.0  0.0   4336   764 ?        Ss   20:36   0:00 /bin/sh -c node server.js
2000         8  0.1  0.5 772124 22604 ?        Sl   20:36   0:00 node server.js
...
```

Saia do seu shell:

```shell
exit
```

## Defina recursos para um Contêiner

Com [`Linux Capabilities`](https://man7.org/linux/man-pages/man7/capabilities.7.html),

> Você pode conceder certos privilégios a um processo, sem conceder todos os privilégios

Reviewer 3: typo: ", Você"

{{< /note >}}

## Defina o perfil do Seccomp para um Contêiner

Para definir o perfil do Seccomp para um contêiner, inclua o campo `seccompProfile` na seção `securityContext` do seu Pod ou manifesto do contêiner. 
O campo `seccompProfile` é um objeto
[SeccompProfile](/docs/reference/generated/kubernetes-api/{{< param "version">}}/#seccompprofile-v1-core) consistindo de `type` e `localhostProfile`.
As opções válidas para `type` incluem `RuntimeDefault`, `Unconfined`, e
`Localhost`. O campo `localhostProfile` só deve ser definido quando `type: Localhost`. 
Este campo indica o caminho do perfil pré-configurado no nó, relativo à localização do perfil de Seccomp do Kubelet 
(configurado com a flag `--root-dir`).

Aqui está um exemplo que define o perfil Seccomp ao perfil padrão do nó do agente de execução do contêiner:

```yaml
...
securityContext:
  seccompProfile:
    type: RuntimeDefault
```

Aqui está um exemplo que define o perfil Seccomp para um arquivo pré-configurado em
`<kubelet-root-dir>/seccomp/my-profiles/profile-allow.json`:

```yaml
...
securityContext:
  seccompProfile:
    type: Localhost
    localhostProfile: my-profiles/profile-allow.json
```

> ## Atribua rótulos SELinux a um contêiner

Reviewer 3: What do you think about keeping the "label" instead of "rótulo"? On line 433 the example uses "label" and "rótulo" can be confusing.

In other docs the word "label" was kept
